openfisca_aotearoa/variables/acts/social_security/supported_living_payment/supported_living_payment__benefit.py

```diff
@@ -0,0 +1,304 @@
+        #     (b) disregard all of the income of a totally blind beneficiary earned by the beneficiary’s own efforts.
+        gross_principal_income = population("social_security__income", period)
+        principal_blind = population("totally_blind", period)
+        principal_disabled = population("supported_living_payment__restricted_work_capacity", period)
+        principal_abled = numpy.logical_not(numpy.logical_or(principal_blind, principal_disabled))
+        principal_disabled_income = gross_principal_income - (20 * period.size_in_weeks)
+        principal_limits = {
+            # if a beneficiary is receiving SLP for blindness, disregard all income
+            principal_blind: numpy.clip(gross_principal_income, 0, 0),
+            # if a beneficiary is receiving SLP for disability, disregard a maximum of $20
+            principal_disabled: numpy.clip(principal_disabled_income, 0, principal_disabled_income),
+            # otherwise, use the gross principal income
+            principal_abled: gross_principal_income}
+        # income based on whether principal population members are blind, disabled, or neither
+        assessable_principal_income = numpy.select(list(principal_limits.keys()), list(principal_limits.values()))
+
+        gross_partner_income = population.partner("social_security__income", period)
+        partner_blind = population.partner("totally_blind", period)
+        partner_disabled = population.partner("supported_living_payment__restricted_work_capacity", period)
+        partner_abled = numpy.logical_not(numpy.logical_or(partner_blind, partner_disabled))
+        partner_disabled_income = gross_partner_income - (20 * period.size_in_weeks)
+        partner_limits = {
+            # if a beneficiary is receiving SLP for blindness, disregard all income
+            partner_blind: numpy.clip(gross_partner_income, 0, 0),
+            # if a beneficiary is receiving SLP for disability, disregard a maximum of $20
+            partner_disabled: numpy.clip(partner_disabled_income, 0, partner_disabled_income),
+            # otherwise, use the gross principal income
+            partner_abled: gross_partner_income}
+        # income based on whether partner population members are blind, disabled, or neither
+        assessable_partner_income = numpy.select(list(partner_limits.keys()), list(partner_limits.values()))
+
+        # numpy.floor required for income tests; reductions are per whole dollar
+        return numpy.floor(assessable_principal_income + assessable_partner_income)
+
+
+class supported_living_payment__reduction(variables.Variable):
+    value_type = float
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "The amount the base benefit is reduced base on the appropriate Income Test and the person & their partners income"
+    reference = "https://legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6784861"
```

Comment:
I wonder if we should update the contributing documentation to cover references in more detail. As a reviewer it would be very helpful for the reference to include each of the versions that inspire the formulas:
That is we use the "latest" link until such time as the logic is changed and then we switch to using the earliest versions the logic applies to, e.g. in this situation:
 https://www.legislation.govt.nz/act/public/2018/0032/56.0/DLM6784861.html (2018 11 26)
 https://www.legislation.govt.nz/act/public/2018/0032/140.0/DLM6784861.html (2020 11 09)